### DISCUSSION TITLE
Pagination with large tables

### DISCUSSION THREAD

**Discussion by Loris-F** [Budibase/budibase#5655] [Help] [answered]
Hello, I am new to Budibase. Just get stuck in some situations so I will ask for some help.
Here a situation:  using Budibase on self-host VPS with with Docker. Installed last version yesterday.
VPS with Debian 11, database Postgres v12.
Try to write a simple app with repeater + table. The problem is about the big table.
The table to show is about 10000 records with many fields. When try to connect the data provider everything get stuck.
Every click on the property is very slow or go to timeout.
There is a correct method to activare the pagination on data provider or maybe in data tables before using it in editing the app?
I have just understand that only table have paginatino, so I can't use query in data source to limit the number of record to show.
Thanks for help.

**Comment by upgreydd** [comment]
In data provider select "pagination" option and provide number of records per page, then in table you'll receive paginated results
![image](https://user-images.githubusercontent.com/579481/165948399-7c8b5815-bc00-4928-ae84-8c0cce7cef0a.png)

**Comment by Loris-F** [comment, marked as answer]
Yes, thats function.
But got ten minutes to click and write that number.
Just to complete the solution for others:
Before setting the "Data" it is better set "Limit"

![immagine](https://user-images.githubusercontent.com/63682343/165950903-a46d7710-d265-4d7f-bef1-b5968e2d3097.png)
